mozci/task.py

```diff
@@ -261,34 +261,56 @@ def _load_errorsummary(self):
         except IndexError:
             return
 
-        groups = None
+        group_results = {}
 
         lines = [json.loads(l) for l in self.get_artifact(path).splitlines()]
+
+        has_group_result = any(line["action"] == "group_result" for line in lines)
+
         for line in lines:
             if line["action"] == "test_groups":
                 groups = list(set(line["groups"]) - {"default"})
 
-            elif line["action"] == "test_result":
+            # After enough time has passed since https://bugzilla.mozilla.org/show_bug.cgi?id=1632242,
+            # we can switch to look for `group_result` lines exclusively.
+            elif not has_group_result and line["action"] == "test_result":
                 group = line.get("group")
                 if group == "default":
                     continue
 
-                self._results.append(
-                    GroupResult(group=group, ok=line["status"] == line["expected"],)
-                )
+                # The "OK" case should never happen given how errorsummary.log works, but
+                # better to be safe than sorry.
+                if line["expected"] == line["status"]:
+                    if group not in group_results:
+                        group_results[group] = "OK"
+                else:
+                    group_results[group] = "ERROR"
+
+            elif line["action"] == "group_result":
+                group_results[line["group"]] = line["status"]
 
             elif line["action"] == "log":
                 self._errors.append(line["message"])
 
+        self._results = [
+            GroupResult(group, True if result == "OK" else False)
+            for group, result in group_results.items()
+            if result != "SKIP"
+        ]
+
         if groups is not None:
             # Assume all groups for which we have no results passed.
-            # In practice, they could also have been skipped (we should ignore them when
-            # it will be feasible, https://github.com/mozilla/mozci/issues/166).
-            known_results = {result.group for result in self._results}
+            # After enough time has passed since https://bugzilla.mozilla.org/show_bug.cgi?id=1632242, we
+            # can stop assuming as all groups will have an associated result.
+            if has_group_result:
+                assert set(groups) == set(
+                    group_results
+                ), f"There are some groups with no results in task {self.id}"
```

Comment:
I'm not sure why, if I inject an `assert False` here, the exception is eaten somewhere along the chain (I confirmed the line was hit by adding print statements before/after the `assert False` line).
Instead, if I add an `assert False` in the caller (`groups`) the exception bubbles up.